File: src/graph_mcp/tools/profile_tools.py
from graph_mcp._select_fields import USER_PROFILE_FIELDS
from graph_mcp.graph_client import graph_client
from graph_mcp.responses import require_auth, success_response


def register_profile_tools(mcp):
    @mcp.tool()
    @require_auth
    async def graph_get_profile() -> str:
        """Get the authenticated user's Microsoft 365 profile."""
        result = await graph_client.get(
            "/me", params={"$select": USER_PROFILE_FIELDS}
        )
        return success_response(result)

File: src/tools/profile-tools.ts
import type { McpServer } from "@modelcontextprotocol/sdk/server/mcp.js";

import { successResponse } from "../responses.js";
import { USER_PROFILE_FIELDS } from "../select-fields.js";
import { registerAuthenticatedTool, type ToolDependencies } from "./tool-types.js";

export function registerProfileTools(
  server: Pick<McpServer, "registerTool">,
  dependencies: ToolDependencies,
): void {
  registerAuthenticatedTool(
    server,
    "graph_get_profile",
    {
      description: "Get the authenticated user's Microsoft 365 profile.",
      inputSchema: {},
    },
    async () => {
      const result = await dependencies.graphClient.get("/me", {
        $select: USER_PROFILE_FIELDS,
      });
      return successResponse(result);
    },
  );
}
